src/sentry/relay/config/experimental.py

```diff
@@ -0,0 +1,88 @@
+import logging
+import time
+from collections.abc import MutableMapping
+from typing import Any, Protocol
+
+import sentry_sdk
+
+logger = logging.getLogger(__name__)
+
+
+class TimeoutException(BaseException):
```

Comment:
[`TimeoutError`](https://docs.python.org/3/library/exceptions.html#TimeoutError) is for system operations.

Reply:
```suggestion
class TimeoutException(Exception):
```

> programmers are encouraged to derive new exceptions from the [Exception](https://docs.python.org/3/library/exceptions.html#Exception) class or one of its subclasses, and not from [BaseException](https://docs.python.org/3/library/exceptions.html#BaseException).

https://docs.python.org/3/library/exceptions.html#built-in-exceptions